src/components/Select/SelectBase.tsx

```diff
@@ -0,0 +1,273 @@
+import * as React from 'react';
+import Downshift, { GetItemPropsOptions, GetToggleButtonPropsOptions } from 'downshift';
+import bem from '../../utils/bem';
+import FieldWrapper from '../FieldWrapper';
+import { List } from '../List';
+import { ENTER_KEY } from '../../constants';
+import { Props as ListRendererProps } from './SuggestionsList';
+import styles from './SelectBase.scss';
+
+export interface Props extends React.HTMLAttributes<HTMLDivElement> {
+    /** an array of objects that will be used to render the suggestions list. */
+    suggestions: object[];
```

Comment:
I think it makes sense to put some effort here and make `Props` interface a generic one. It would allow us to express type logic better and probably catch some errors in the future.

It would looks like:

```ts
interface Props<S> extends React.HTMLAttributes<HTMLDivElement> {
    /** an array of objects that will be used to render the suggestions list. */
    suggestions: S[];
    suggestionToString: (suggestions?: S | null) => string;
    onSelectionChange: (item: S) => void;
```

Reply:
This is really good. I didn't know how to do this, hence I used this object thing

Reply:
There is [an article about JSX and generics](https://mariusschulz.com/blog/passing-generics-to-jsx-elements-in-typescript) that I googled yesterday, can be also helpful